backend/tracim_backend/lib/core/user.py

```diff
@@ -1121,3 +1123,99 @@ def allowed_to_invite_new_user(self, email: str) -> bool:
             return False
 
         return True
+
+    def create_user_following(
+        self,
+        follower_id: int,
+        leader_id: int,
+        created_date: typing.Optional[datetime.datetime] = None,
+        do_save: bool = True,
+    ) -> UserFollower:
+        if (
+            self._session.query(UserFollower)
+            .filter(
+                and_(UserFollower.leader_id == leader_id, UserFollower.follower_id == follower_id)
+            )
+            .count()
+        ):
+            raise UserFollowAlreadyDefined("User follow already defined")
```

Comment:
Shouldn't you let the database do this check using a unique constraint? This would make the operation atomic.

Reply:
I'm not sure than sqalchemy raise an specific exception for that. I check and use it if it is the case.

Reply:
I confirm that: When using sqlite, sqlalchemy raise `sqlite3.IntegrityError`. This is database specific. In addition, error is raised at commit, so **after** view execution (so we can't catch it with hapic view decorators)

Reply:
Maybe i ca explai that in comment ?

Reply:
I juste added it in comment